jupyter_elfinder/elfinder.py

```diff
@@ -1200,15 +1206,28 @@ def __ls(self) -> None:
             self._response[R_ERROR] = "Target directory not found"
             return
 
-        items = {}
-        for fname in os.listdir(path):
-            fhash = self.__hash(os.path.join(path, fname))
-            if intersect:
-                if fhash in intersect:
+        if os.path.islink(path):
+            path = self.__read_link(path)
+            if path is None:
+                self._response[R_ERROR] = "Directory (link) not found"
+                return
+
+        if not self.__is_allowed(path, "read"):
+            self._response[R_ERROR] = "Access denied"
+            return
+
+        try:
+            items = {}
+            for fname in os.listdir(path):
```

Comment:
It's easier to read if we limit the scope of the try... except as much as possible by only wrapping the line(s) that can raise. We also save on indentation.

```py
items = {}
try:
    file_names = os.listdir(path)
except PermissionError:
    self._response[R_ERROR] = "Access denied"
    return
for fname in file_names:
    ...
```